docs/sources/query/bp-query.md

```diff
@@ -0,0 +1,80 @@
+---
+title: Query best practices
+menuTitle:  Query best practices
+description: Describes best practices for querying in Grafana Loki.
+aliases:
+- ../bp-query
+weight: 700
+---
+# Query best practices
+
+The way you write queries in Loki affects how quickly you get results returned from those queries. Understanding the way Loki parses queries can help you write queries that are efficient and performant.
+
+{{< admonition type="tip" >}}
+Before you start optimizing queries, read the [labels best practices]({{< relref "../get-started/labels/bp-labels" >}}) page to understand what makes a good label. Choosing the right labels in the first place is the first step towards writing efficient queries.
+{{< /admonition >}}
+
+Loki evaluates a LogQL query from left to right, in the order that it is written. To get the best possible query performance, eliminate as many potential results as you can earlier in the query and then continue to progressively narrow your search as you continue writing the query. On this page is the recommended order for how to filter efficiently in your query.
+
+## Narrow down your time range first
+
+Reduce the number of logs Loki needs to look through by specifying a period of time that you'd like to search through. Loki creates one index file per day, so queries that span over multiple days fetches multiple index files. The fewer files are searched, the faster the query results are returned.
+
+Time ranges are typically not part of the query, but you can set a time range through your visualization tool or through [the Loki API]({{< relref "../reference/loki-http-api/" >}}).
+
+
+### In Grafana
+
+If you're using Loki with Grafana, you can use the dropdown menu on the upper right hand corner of a dashboard to select a time range, either relative (last X hours) or absolute (a specific date and time).
+
+![Screenshot of time selector on Grafana](../grafana-time-range-picker.png "Grafana time interval selector")
+
+### Through Loki API
+
+If you're querying Loki through [the Loki API]({{< relref "../reference/loki-http-api/" >}}), you can use the [`query_range` endpoint]({{< relref "../reference/loki-http-api/#query-logs-within-a-range-of-time" >}}) to add `start` and `end` timestamps for your query as parameters to the HTTP call rather than as part of the query itself.
+
+```bash
+http://<loki-instance>/loki/api/v1/query_range?query={job="app"}&start=1633017600000000000&end=1633104000000000000
+
+```
+
+
+## Use precise label selectors
+
+Next, write your label selectors. Identify the most specific label you can use within the log line and search based on that first. For example, if the logs contain the labels `org_id` and `employee_uid` and the latter is a smaller subset of data, start your query by selecting based on `employee_uid`:
+
+```bash
+{employee_uid="PGJ1Fr4Zz"}
```

Comment:
Not really a label you should be using, I would vote for something more workload related namespace, cluster, app_name, service_name.

Reply:
You're right in hindsight. Changed!